cpp/connectivity_module/connectivity_module.cpp

```diff
@@ -1,136 +1,95 @@
-       vertex != nullptr;
-       vertex = mgp_vertices_iterator_next(vertices_iterator)) {
-    mgp_vertex_id vertex_id = mgp_vertex_get_id(vertex);
-    if (vertex_component.find(vertex_id.as_int) != vertex_component.end())
-      continue;
-
-    // Run BFS from current vertex.
-    std::queue<int64_t> q;
-    q.push(vertex_id.as_int);
-    vertex_component[vertex_id.as_int] = curr_component;
-    while (!q.empty()) {
-      mgp_vertex *v = mgp_graph_get_vertex_by_id(graph, {q.front()}, memory);
-      if (v == nullptr) {
-        mgp_result_set_error_msg(result, "Unable to get vertex");
-        return;
-      }
-      utils::OnScopeExit v_deleter([&v] {
-        if (v != nullptr) {
-          mgp_vertex_destroy(v);
-        }
-      });
+  try {
+    mg_graph::Graph<> *graph =
+        mg_utility::GetGraphView(memgraph_graph, result, memory);
 
-      q.pop();
+    std::unordered_map<uint64_t, uint64_t> vertex_component;
+    int64_t curr_component = 0;
+    for (mg_graph::Node vertex : graph->Nodes()) {
+      if (vertex_component.find(vertex.id) != vertex_component.end())
+        continue;
 
-      // Iterate over inbound edges.
-      mgp_edges_iterator *edges_iterator = mgp_vertex_iter_in_edges(v, memory);
-      if (edges_iterator == nullptr) {
-        mgp_result_set_error_msg(result, "Unable to create edges iterator");
-        return;
-      }
-      for (const mgp_edge *edge = mgp_edges_iterator_get(edges_iterator);
-           edge != nullptr; edge = mgp_edges_iterator_next(edges_iterator)) {
-        mgp_vertex_id next_id = mgp_vertex_get_id(mgp_edge_get_from(edge));
-        if (vertex_component.find(next_id.as_int) != vertex_component.end())
-          continue;
-        vertex_component[next_id.as_int] = curr_component;
-        q.push(next_id.as_int);
-      }
+      // Run BFS from current vertex.
+      std::queue<uint64_t> q;
 
-      // Iterate over outbound edges.
-      mgp_edges_iterator_destroy(edges_iterator);
-      edges_iterator = mgp_vertex_iter_out_edges(v, memory);
-      if (edges_iterator == nullptr) {
-        mgp_result_set_error_msg(result, "Unable to create edges iterator");
-        return;
-      }
-      for (const mgp_edge *edge = mgp_edges_iterator_get(edges_iterator);
-           edge != nullptr; edge = mgp_edges_iterator_next(edges_iterator)) {
-        mgp_vertex_id next_id = mgp_vertex_get_id(mgp_edge_get_to(edge));
-        if (vertex_component.find(next_id.as_int) != vertex_component.end())
-          continue;
-        vertex_component[next_id.as_int] = curr_component;
-        q.push(next_id.as_int);
-      }
+      q.push(vertex.id);
+      vertex_component[vertex.id] = curr_component;
+      while (!q.empty()) {
+        uint64_t v_id = q.front();
+        q.pop();
 
-      mgp_edges_iterator_destroy(edges_iterator);
-    }
+        // Iterate over inbound edges.
+        for (mg_graph::Neighbour neihgbor : graph->Neighbours(v_id)) {
+          uint64_t next_id = neihgbor.node_id;
```

Comment:
prefer `auto` so you don't have to search and replace when you change the underlying type.  :smile: